src/ucp/wireup/wireup.c

```diff
@@ -293,20 +293,20 @@ ucp_wireup_find_remote_p2p_addr(ucp_ep_h ep, ucp_lane_index_t remote_lane,
 }
 
 static ucp_lane_index_t
-ucp_wireup_ep_configs_can_reuse_lane(const ucp_ep_config_key_t *key1,
-                                     const ucp_ep_config_key_t *key2,
-                                     ucp_lane_index_t lane)
+ucp_wireup_ep_configs_find_reuse_lane(const ucp_ep_config_key_t *key1,
+                                      ucp_lane_index_t lane1,
+                                      const ucp_ep_config_key_t *key2)
```

Comment:
seems like we need typedef for currently anonymous struct inside key, then it could be like
```
static ucp_lane_index_t
ucp_ep_config_key_lane_by_attrs(const ucp_ep_config_key_t *key, const ucp_ep_config_lane_attrs_t *attrs)
```
i guess it could simplify other places

Reply:
do you mean to replace `ucp_wireup_ep_configs_find_reuse_lane()` by `ucp_ep_config_key_lane_by_attrs()`, but I don't understand how will it work

Reply:
for example from your code if i don't mess indexes...
```
        intersect_map[lane1_idx] =
            ucp_ep_config_key_lane_by_attrs(key2, &key1->lanes[lane1_idx]);
```

Reply:
unfortunately, it's not possible `ucp_ep_config_find_match_lane()` has to know the `lane1` number to report it as a result in case of CM or sockaddr stub lane
and `ucp_ep_config_lane_is_same_peer()` requires two keys ant lanes

Reply:
actually, with new typedef
`ucp_ep_config_lane_is_same_peer` could be `ucp_ep_config_lane_attrs_peer_is_equal(lane_attrs1, lane_attrs2)`
and CM/sockaddr stub lane cases have to be handled separately

Reply:
still it will be impossible, since we have `ucp_ep_config_lane_is_proxy_equal()` that requires `key1`/`key2` to check proxy lane
I'd remain it as is, and evaluate it then

Reply:
just fixed the names of the functions as you suggested

Reply:
> still it will be impossible, since we have `ucp_ep_config_lane_is_proxy_equal()` that requires `key1`/`key2` to check proxy lane
> I'd remain it as is, and evaluate it then

As I see proxy lane is part of "arrt" `key2->lanes[lane2].proxy_lane`

Reply:
yes, but we won't be able to get attributes for `proxy_lane` if we pass only attributes for the original `lane`

Reply:
to check proxy attrs it could be separate call with proxy lanes attrs in args